### ISSUE TITLE
All records received during initial snapshot must have zero Uint64(0) _version

### ISSUE THREAD

**Issue by R-omk** [Altinity/clickhouse-sink-connector#800]
.. now there is a timestamp ms  plus  a serial number,      but this value may be greater than the value calculated during  cdc event processing, especially when the `snowflake.id`  is disabled.

![image](https://github.com/user-attachments/assets/ae607473-e9dc-4721-b942-4d061d5da733)

in the example on the screenshot the version is calculated from GTID position, it is obvious that the  ReplacingMergeTree engine  cannot works properly

---

context: 
version: 2.3.0 
``snowflake.id: "false"``

**Comment by R-omk** [comment]
I would also like to note that snowflake id values ​​are only comparable between values ​​of the same type of snowflake id; they cannot be compared with an arbitrary timestamp.      

... especially considering that a custom time offset is used for snowflakeid generator.

https://github.com/Altinity/clickhouse-sink-connector/blob/1027f19af3e94dd1638d7dc986870b3b6842987a/sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/common/SnowFlakeId.java#L16

**Comment by aadant** [comment]
Yes I agree it should be zero. You get that it you use the python utilities to dump the data.
It is not an issue if the _version is always increasing. It should always be the case.
